packages/player/src/machines/video-machine.ts

```diff
@@ -51,262 +71,402 @@ export interface VideoStateContext {
+            target: 'ready',
+            cond: {
+              type: 'resourceLoaded',
+            },
+            actions: assign({
+              readyState: (context, _event) =>
+                context.videoRef?.current?.readyState ?? 0,
+              waiting: (_context, _event) => false,
             }),
-            'seekVideo',
-          ],
+          },
+          FAIL: 'failure',
         },
-        END_SEEKING: {
-          actions: [
-            assign({
-              seekingTime: (context, event) => 0,
-              seeking: (_context, _event) => false,
-            }),
-          ],
+      },
+      ready: {
+        initial: 'paused',
+        entry: [
+          assign({
+            waiting: (_context, _event) => false,
+            isSubmittingCueNote: (_context, _event) => false,
+          }),
+        ],
+        on: {
+          TAKE_NOTE: {
+            target: 'taking_note',
+          },
+          VOLUME_CHANGE: {
+            actions: [
+              assign({
+                volume: (_context, event) => event.volume,
+              }),
+              'setVolume',
+            ],
+          },
+
+          TOGGLE_MUTE: {
+            actions: ['toggleMute'],
+          },
+          PLAYBACKRATE_CHANGE: {
+            actions: [
+              assign({
+                playbackRate: (_context, event) => event.playbackRate,
+              }),
+              'setPlaybackRate',
+            ],
+          },
+        },
+        states: {
+          paused: {
+            on: {
+              PLAY: {
+                target: 'playing',
+                actions: [
+                  assign({
+                    hasStarted: (_context, _event) => true,
+                    lastAction: (_context, _event) => 'PLAY',
+                  }),
+                  'playVideo',
+                ],
+              },
+            },
+          },
+          playing: {
+            entry: [
+              assign({
+                waiting: (_context, _event) => false,
+              }),
+            ],
+            on: {
+              PAUSE: {
+                target: 'paused',
+                actions: [
+                  'pauseVideo',
+                  assign({
+                    lastAction: (_context, _event) => 'PAUSE',
+                  }),
+                ],
+              },
+              TIMING: {
+                actions: assign({
+                  currentTime: (context, _event) =>
+                    context.videoRef?.current?.currentTime ?? 0,
+                  duration: (context, _event) =>
+                    context.videoRef?.current?.duration ?? 0,
+                }),
+              },
+              END: 'ended',
+            },
+          },
+          ended: {
+            entry: ['onVideoEnded'],
+            on: {PLAY: 'playing'},
+          },
         },
       },
-      states: {
-        paused: {
-          on: {
-            PLAY: {
-              target: 'playing',
+      taking_note: {
```

Comment:
in order to get the authoring notes flow just right UX wise, I needed to distinguish between focused, typing, and submitting states.  state machine with `taking_note` state works a little like this:

![state machine gif](https://user-images.githubusercontent.com/25487857/149932401-e4016ed8-d253-4f8e-8d5d-89396ac2b8ae.gif)